### PR TITLE
Filter improvements

### PR DESCRIPTION
**Pull request by Lathrisk**
#113
#94

### CODE REVIEW

**Comment by Lathrisk** [comment]
This updates the back to all publishers copy on the publisher page nav button.

360-back-to-all

And adds a multi-select filter box for all publishers using the vue-select library. A clear all button has also been added that removes all selections.

360-filters

**Review by michaelwood — CHANGES_REQUESTED** [review]
All the code looks good to me, one minor thing could we change `publisherParams` to something more friendly as we're exposing that in the url.

As we're now changing the filter functionality to be on the client side and we're not just rendering the output I think we're now in the "really need a selenium test" threshold. vue select looks well tested so only looking for a basic test for the code in the app.

**Comment by Lathrisk** [comment]
@michaelwood I've updated the URL parameter to just `publishers`, while keeping publisherParams in the code.

In terms of tests. I suspect we want to separate the vue testing from the flask testing in case we want to split them in future, so is selenium the best choice for that? It looks like nightwatch.js is the bundled e2e testing tool for vue-cli. There are other options too, but if we want to keep testing consistency with other parts of 360 then we can use selenium.

File: tests_dashboard/test_dashboard.py [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+def test_default_titles(browser):
+    browser.get(url_for('dashboard', _external=True) + 'publishers/')
+    title = browser.find_elements(By.CLASS_NAME, 'dashboard-publisher-result__title')
+    assert title[0].get_attribute("innerHTML") == ' A B Charitable Trust '
+    assert title[1].get_attribute("innerHTML") == ' Access to Justice Foundation '
+
+
+@pytest.mark.usefixtures('live_server')
+def test_filtered_titles(browser):
+    browser.get(url_for('dashboard', _external=True) + 'publishers?publisherParams=360G-BNG&publisherParams=360G-coopgroup')
```
Comment (michaelwood) [inline tests_dashboard/test_dashboard.py:39]: Is publisherParams correct here? Given the change in the url schema to publisher= ?

**Review by michaelwood — CHANGES_REQUESTED** [review]
One comment on the Is publisherParams correct here: https://github.com/ThreeSixtyGiving/registry/blob/filter-improvements/tests_dashboard/test_dashboard.py#L39 . If you can squash the vue rebuilds before you merge it that'd be nice too.

File: tests_dashboard/test_dashboard.py (hunk 1 above)
Comment (Lathrisk) [reply tests_dashboard/test_dashboard.py:39]: Yeah, the base is old, updating now.

File: tests/test_functional.py [hunk 2]
```diff
@@ -4,6 +4,7 @@
 from flask import url_for
 from selenium import webdriver
 from selenium.webdriver.chrome.options import Options
+from selenium.webdriver.common.by import By
```
Comment (michaelwood) [inline tests/test_functional.py:7]: :+1:

**Review by michaelwood — APPROVED** [review]

**Pull request merged**